app/models/miq_report_result.rb

```diff
@@ -142,8 +136,12 @@ def self.parse_userid(userid)
     raise _("Cannot parse userid %{user_id}") % {:user_id => userid.inspect}
   end
 
+  def self.purge_for_all_users
+    purge_for_user(:userid => "%")
+  end
+
   def self.purge_for_user(options = {})
-    options[:userid] ||= "%" # This will purge for all users
+    raise ArgumentError, "options must contain a :userid" unless options[:userid]
```

Comment:
Technically, this is "new", but there aren't any callers that pass a missing userid, *except* for the atStartup one.  It seemed like a dangerous action waiting to happen, so I split the logic.